Diplom_1 pull_request

Mswahili88 opened the pull request:
Created the tests for classes Bun, Ingredient, Database and Burger.  Covering of 70%+ is done.
Were used such libraries as pytest, pytest -cov and coverage. Made extra file data.py and conftest.py

Mswahili88 committed:
Create the tests for classes Bun, Ingredient and Burger.

Mswahili88 committed:
Added tests for class Database and took back report from .ignore file

Mswahili88 committed:
trying took back report from .ignore file

Mswahili88 committed:
trying took back report from .ignore file again, deleting file .ignore from folder with report

AlexKitIt commented on conftest.py:
```diff
@@ -0,0 +1,27 @@
+
+    return ingredient
+
+@pytest.fixture()
+def file_o_fish():
+    return Burger()
+
+@pytest.fixture()
+def database():
+    return Database()
```
Нужно исправить: фикстуры они выполняют логику предусловий\постусловий а не создают рандомные объекты, которые создаются в тестах

AlexKitIt commented on tests/test_burger.py:
```diff
@@ -0,0 +1,33 @@
+        file_o_fish.add_ingredient(mock_ingredient)
+        file_o_fish.remove_ingredient(0)
+        assert file_o_fish.ingredients == []
+
+
+
+
```
Нужно исправить: не проверены методы move_ingridient, get_recipe

AlexKitIt commented on tests/test_ingredient.py:
```diff
@@ -0,0 +1,16 @@
+
+    def test_get_price_ingredient(self):
+        ingredient = Ingredient(ingredient_types.INGREDIENT_TYPE_SAUCE, IngrData.ingr_name, IngrData.ingr_price)
+        assert ingredient.get_price() == IngrData.ingr_price and ingredient.get_type() == 'SAUCE'
```
Нужно исправить: судя по названию это проверка метода get_price, почему тут в assert еще и .get_type?

Mswahili88 committed:
1. Got rid of false fixtures
2. Added two test_methods despite my coverage wa 70% already
3. And corrected one assert in get_price method
